### PR TITLE
Add resource limits and requests for the interceptor pod component

### LINKED ISSUE #420: keda-hpa-keda-add-ons-http-interceptor unable to get cpu utilization
keda-hpa-keda-add-ons-http-interceptor is created when you install the http add on, the HPA is to scale the http interceptor itself.
However, since the deployment does not contain any requests/limits, the HPA throws a `FailedGetResourceMetric` event with a message of `failed to get cpu utilization: missing request for cpu`

## Expected Behavior
The HPA scales correctly for the interceptor when it is under high cpu load

## Actual Behavior
No scaling occurs due to the HPA being unable to read cpu utilization

## Steps to Reproduce the Problem

  1. Deploy http add ons
  2. Deploy the sample app per http add ons' directions
  3. run `kubectl describe hpa keda-hpa-keda-add-ons-http-interceptor` on your hpa and you will see that the `ScalingActive` condition is `False`
  4. Edit or patch the deployment of the interceptor and add limits/requests (or re-run helm setting resource values for interceptor)
  5. run `kubectl describe hpa keda-hpa-keda-add-ons-http-interceptor` and ensure that the `ScalingActive` is `True`

## Specifications

  - **KEDA Version:** 2.8.2
  - **Platform & Version:** amd64
  - **Kubernetes Version:** K8s 1.23
  - **Scaler(s):** http add on version 0.4.0 installed via chart 0.4.1

### PR DESCRIPTION
**Pull request by chrisjohnson00**
This resolves issues #420 by adding default resource limits/requests

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*
- [x] A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*
  - n/a

Fixes #420

### CODE REVIEW

**Comment by chrisjohnson00** [comment]
I've also created PR #423 which fixes the CI failure seen in this (unrelated to this PR's change)

File: http-add-on/values.yaml [hunk 1]
```diff
@@ -122,6 +122,14 @@ interceptor:
   nodeSelector: {}
   tolerations: []
   affinity: {}
+  # interceptor pod resource limits
+  resources:
+    limits:
+      cpu: 0.5
```
Comment (JorTurFer) [inline http-add-on/values.yaml:128]: Why not `500m`? Am I missing something?

File: http-add-on/Chart.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ kubeVersion: ">=v1.23.0-0"
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.4.1
+version: 0.4.2
```
Comment (JorTurFer) [inline http-add-on/Chart.yaml:13]: Please revert this change. We update the version during releases

**Review by JorTurFer — COMMENTED** [review]
Thanks for the improvement, I have left some inline comments.

@tomkerkhove , what do you think about the values?

**Comment by tomkerkhove** [comment]
Out of curiosity - How did you determine the numbers?

File: http-add-on/Chart.yaml (hunk 2 above)
Comment (chrisjohnson00) [reply http-add-on/Chart.yaml:13]: Ah, thanks!  I will revert that and add to the comment stating such, as the comment led me to believe I had to increment it myself.

**Comment by chrisjohnson00** [comment]
> Out of curiosity - How did you determine the numbers?

They are a copy/paste from the operator values.  Also, when I ran some load against my cluster, I never saw metrics report the CPU limited.  If you have better ideas (like I'd rather not limit CPU to < 1), please let me know and I'll be happy to make the changes.

**Review by JorTurFer — APPROVED** [review]
LGTM!
Let's wait @tomkerkhove review 😄

**Comment by JorTurFer** [comment]
BTW, we need that all the commits are signed, could you sign them? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

**Comment by chrisjohnson00** [comment]
> BTW, we need that all the commits are signed, could you sign them? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

Oop, good catch, I've amended my last commit message.

**Comment by tomkerkhove** [comment]
> BTW, we need that all the commits are signed, could you sign them? https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#i-didnt-sign-my-commit-now-what

Signed off, not signed - a very confusing difference 😅

**Comment by tomkerkhove** [comment]
> > Out of curiosity - How did you determine the numbers?
> 
> They are a copy/paste from the operator values.  Also, when I ran some load against my cluster, I never saw metrics report the CPU limited.  If you have better ideas (like I'd rather not limit CPU to < 1), please let me know and I'll be happy to make the changes.

Fair enough, we can adapt later on as well!

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
Thanks!
